components/nimbus/src/metrics.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{
+    enrollment::ExperimentEnrollment, glean_metrics::nimbus_events::EnrollmentStatusExtra,
+    EnrollmentStatus,
+};
+use serde_derive::{Deserialize, Serialize};
+
+pub trait HostMetricsHandler: Send + Sync {
+    fn record_enrollment_status(&self, enrollment_status_extra: EnrollmentStatusExtraDef);
+
+    /// Only used for testing
+    fn get_enrollment_status_records(&self) -> Vec<EnrollmentStatusExtraDef>;
+
+    /// Only used for testing
+    fn clear(&self);
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(remote = "EnrollmentStatusExtra")]
```

Comment:
This is nice because even though we have to manually re-define the struct here, the `serde-remote` piece does at verify that they have the same members.